src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -43,6 +44,16 @@ def status(self):
             status['message'] = str(ex)
         return status
 
+    @staticmethod
+    def handle_error(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except (DashboardException, RequestException) as e:
+                raise DashboardException(e, http_status_code=500, component='rgw')
```

Comment:
In this particular case, to keep the current value relying on the class, we should go for:
```
component=Rgw.__name__.lower()
```
Is this preferable?

Reply:
What I mean is that if you embed this function into a class is just to benefit somehow from that class? If not, I think it's better to move that out of the class scope. Personally, if we want to print the component name, rather than explicitly setting that for every component I'd try to 'infer' it (by using `class.__name__`). It's less beautiful than writing a custom name for every component, but it works with everything.